src/vs/workbench/contrib/terminal/browser/terminalInstance.ts

```diff
@@ -1324,7 +1323,17 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 		}
 		const Addon = await this._terminalInstanceService.getXtermWebglConstructor();
 		this._webglAddon = new Addon();
-		this._xterm.loadAddon(this._webglAddon);
+		try {
+			this._xterm.loadAddon(this._webglAddon);
+		} catch {
+			this._disposeOfWebglRenderer();
+			this._safeSetOption('rendererType', 'dom');
```

Comment:
We'll want to remember this failure and not try again, I guess until the setting changes so it's possible to fix it up if for example hardware changes in the meantime by changing the setting?